runtime/compiler/env/JITServerCHTable.cpp

```diff
@@ -0,0 +1,664 @@
+            innerAssumptions.push_back(innerInfo);
+            }
+         }
+
+      serialVGuards.push_back(std::make_tuple(info, nopSites, innerAssumptions));
+      }
+
+   TR::list<TR_VirtualGuardSite*> &sideEffectPatchSites = *comp->getSideEffectGuardPatchSites();
+   std::vector<TR_VirtualGuardSite> sideEffectPatchSitesVec;
+   for (TR_VirtualGuardSite *site : sideEffectPatchSites)
+      sideEffectPatchSitesVec.push_back(*site);
+
+   FlatClassLoadCheck compClassesThatShouldNotBeLoaded;
+   for (TR_ClassLoadCheck * clc = comp->getClassesThatShouldNotBeLoaded()->getFirst(); clc; clc = clc->getNext())
+      compClassesThatShouldNotBeLoaded.emplace_back(std::string(clc->_name, clc->_length));
+
+   FlatClassExtendCheck compClassesThatShouldNotBeNewlyExtended;
+   for (TR_ClassExtendCheck * cec = comp->getClassesThatShouldNotBeNewlyExtended()->getFirst(); cec; cec = cec->getNext())
+      compClassesThatShouldNotBeNewlyExtended.emplace_back(cec->_clazz);
+
+   auto *compClassesForOSRRedefinition = comp->getClassesForOSRRedefinition();
+   std::vector<TR_OpaqueClassBlock*> classesForOSRRedefinition(compClassesForOSRRedefinition->size());
+   for (int i = 0; i < compClassesForOSRRedefinition->size(); ++i)
+      classesForOSRRedefinition[i] = (*compClassesForOSRRedefinition)[i];
+
+   uint8_t *startPC = comp->cg()->getCodeStart();
+
+   return std::make_tuple(classes,
+                          classesThatShouldNotBeNewlyExtended,
+                          preXMethods,
+                          sideEffectPatchSitesVec,
+                          serialVGuards,
+                          compClassesThatShouldNotBeLoaded,
+                          compClassesThatShouldNotBeNewlyExtended,
+                          classesForOSRRedefinition,
+                          startPC);
+   }
+
+// Must hold classTableMonitor when calling this method
+void cleanupNewlyExtendedInfo(TR::Compilation *comp, std::vector<TR_OpaqueClassBlock*> &classesThatShouldNotBeNewlyExtended)
+   {
+   TR_PersistentCHTable *table = comp->getPersistentInfo()->getPersistentCHTable();
+   for (auto classId : classesThatShouldNotBeNewlyExtended)
+      {
+      TR_PersistentClassInfo * cl = table->findClassInfo(classId);
+      // The class may have been unloaded during this compilation and the search may return NULL
+      // This method is called even if we abort the compilation. Hence, killing the
+      // compilation after a class unload (which we already do) is not ehough
+      // If we get really unlucky a new class could be unloaded exactly in the place of the
+      // unloaded one (the window of time is very small for this to occur). However, resetting
+      // the flag for the newly loaded class is not going to create any problems (it's
+      // supposed to be reset to start with)
+      if (cl)
+         cl->resetShouldNotBeNewlyExtended(comp->getCompThreadID());
+      }
+   }
+
+
+
+bool JITServerCHTableCommit(
```

Comment:
If this is performed by the JITClient as I think it is, then it should be renamed to `JITClientCHTableCommit`